### PR TITLE
Fix access key creation

### PR DESCRIPTION
**Pull request by 11xor6**
Added wait for access key to register with AWS without which caused successive calls to fail with:
    AWS Error Code: AuthFailure, AWS Error Message: AWS was not able to validate the provided access credentials

### CODE REVIEW

File: airship-cli/src/main/java/io/airlift/airship/cli/Airship.java [hunk 1]
```diff
@@ -1244,6 +1247,26 @@ private static String createIamUserForEnvironment(AmazonIdentityManagementClient
             String policyJson = JsonCodec.jsonCodec(Object.class).toJson(policy);
 
             iamClient.putUserPolicy(new PutUserPolicyRequest(username, "policy", policyJson));
+
+            // Back-off waiting for the UserPolicy to register with AWS.
+            long wait = 1L;
+            for (int i = 0; i < 8; i++) {
+                System.out.printf("Waiting %s seconds for UserPolicy to register with AWS (attempt #%s)\n", wait, i + 1);
+                try {
+                    iamClient.getLoginProfile(new GetLoginProfileRequest(username));
```
Comment (11xor6) [inline airship-cli/src/main/java/io/airlift/airship/cli/Airship.java]: This is actually not a good test for the availability of the user; I'm working on a different solution.

**Comment by 11xor6** [comment]
Updated the pull request; I found the root cause of the problem and redirected.

**Comment by electrum** [comment]
Some comments after discussing with @dain:
- This code doesn't actually do exponential backoff.
- It's annoying for the user to have to wait 15 seconds every time, even if the credentials start working within a few seconds. Since it's basically free to make the call, we might as well check once per second for two minutes (and don't sleep before the first check). No need for exponential backoff or long delays. Then you can have the message just be `"Waiting for access key to register with AWS..."` with an extra dot per second.
- You'll never see an `InterruptedException` from the CLI, so you simplify the code by using `Uninterruptibles.sleepUninterruptibly()`.

**Comment by 11xor6** [comment]
@electrum I've updated the pull request taking into account the given comments.

**Comment by electrum** [comment]
Looks good, thanks!

**Pull request closed without merging**